oagutu/hellobooks_client
# [FT] Admin Roles

#### What does this PR do?

It adds support for admin functions such as: add/edit book, delete book, access member list, and change member status 
#### Description of Task to be completed?
Given an admin user is logged in, they should be able to access and perform roles reserved only for the admin. These include: add/edit book, delete book, change member status
#### How should this be manually tested?

- A user should log in using an admin account.
- Once logged in they should be able to see and interact with the edit and delete icons for each book entry in the library table eg. to edit a book they can click on the edit icon which should bring up the edit form as a modal.
- The user can then click on the dropdown int he nabber to access roles such as view member list, change member status, add book and view logs.

#### What are the relevant pivotal tracker stories?
[#159890804: Add/Edit Books](https://www.pivotaltracker.com/story/show/159890804)
[#159891030: View Logs](https://www.pivotaltracker.com/story/show/159891030)
[#159890947: Change user status/role](https://www.pivotaltracker.com/story/show/159890947)
[#159891068: Delete book](https://www.pivotaltracker.com/story/show/159891068)

#### Screenshots:
`Dropdown menu showing admin roles`
<img width="1065" alt="screen shot 2018-08-26 at 20 50 01" src="https://user-images.githubusercontent.com/37480490/44631212-acc83280-a971-11e8-9943-34874c6935f5.png">

`Example of edit and delete icons(actions) on each book entry`
<img width="1054" alt="screen shot 2018-08-26 at 20 49 20" src="https://user-images.githubusercontent.com/37480490/44631229-d2edd280-a971-11e8-8a8f-bcb564194cf2.png">

`Add book page`
<img width="1048" alt="screen shot 2018-08-26 at 20 49 08" src="https://user-images.githubusercontent.com/37480490/44631235-e13bee80-a971-11e8-8b7a-fac5d583a48b.png">

`Change user role/status`
<img width="1038" alt="screen shot 2018-08-26 at 20 49 34" src="https://user-images.githubusercontent.com/37480490/44631239-f284fb00-a971-11e8-8309-4df364b5fce3.png">

`View logs`
<img width="1047" alt="screen shot 2018-08-26 at 20 49 49" src="https://user-images.githubusercontent.com/37480490/44631243-fdd82680-a971-11e8-8baf-e545a2b05b66.png">
